core/src/main/resources/common/message/GroupMetadataResponse.json

```diff
@@ -0,0 +1,103 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "name": "GroupMetadataResponse",
+  "validVersions": "0-3",
+  "fields": [
+    {
+      "name": "protocolType",
+      "versions": "0+",
+      "type": "string"
+    },
+    {
+      "name": "generation",
+      "versions": "0+",
+      "type": "int32"
+    },
+    {
+      "name": "protocol",
+      "versions": "0+",
+      "type": "string",
+      "nullableVersions": "0+"
+    },
+    {
+      "name": "leader",
+      "versions": "0+",
+      "type": "string",
+      "nullableVersions": "0+"
+    },
+    {
+      "name": "members",
+      "versions": "0+",
+      "type": "[]MemberMetadata"
+    },
+    {
+      "name": "currentStateTimestamp",
+      "versions": "2+",
+      "type": "int64"
+    }
+  ],
+  "commonStructs": [
+    {
+      "name": "MemberMetadata",
+      "versions": "0-3",
+      "fields": [
+        {
+          "name": "memberId",
+          "versions": "0+",
+          "type": "string"
+        },
+        {
+          "name": "clientId",
+          "versions": "0+",
+          "type": "string"
+        },
+        {
+          "name": "clientHost",
+          "versions": "0+",
+          "type": "string"
+        },
+        {
+          "name": "sessionTimeout",
+          "versions": "0+",
+          "type": "int32"
+        },
+        {
+          "name": "subscription",
+          "versions": "0+",
+          "type": "bytes"
```

Comment:
We may want to use `"zeroCopy": true` for both `subscription` and `assignment` like we did for the consumer protocol.

Reply:
I am not sure if we gain much by using `"zeroCopy": true` in the end. I suggested this because the Struct based schemas were using `ByteBuffer`. Using `bytes` should be ok here, isn't it?

Reply:
> Using bytes should be ok here, isn't it?

It would be better to use bytes currently since ```MemberMetadata.scala``` is using bytes rather than ```ByteBuffer```(https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/coordinator/group/MemberMetadata.scala#L64). The previous code make deep copy of ```ByteBuffer``` from struct to get bytes

```scala
member.assignment = Utils.toArray(memberMetadata.get(ASSIGNMENT_KEY).asInstanceOf[ByteBuffer])
```